cypress/support/helpers/athenaReportHelper.js

```diff
@@ -0,0 +1,117 @@
+import adminPageElements from "../elements/admin_page_elements";
+import signinPageElements from "../elements/signin_page_elements";
+import researcherReportsPageElements from "../elements/researcher_reports_page_elements";
+
+const ATHENA_REPORTS_URL = 'https://learn-report.staging.concord.org/auth/login?after_sign_in_path=%2Freport%2Flearner';
+
+export function loginToAthenaReports(username, password){
+    cy.visit(ATHENA_REPORTS_URL);
+    cy.url().then(url => {
+        if(url.includes('/auth/login')){
+            cy.get(signinPageElements.USERNAME_FIELD_SIGNIN_PAGE).type(username);
+            cy.get(signinPageElements.PASSWORD_FIELD_SIGNIN_PAGE).type(password);
+            cy.get(signinPageElements.SUBMIT_BUTTON_SIGNIN_PAGE).click();
+        }
+    });
+
+}
+
+export function invokeUsageReportAPI(reportAPI, expectedOutputFile){
+    invokeReportAPI(reportAPI, expectedOutputFile, true);
+}
+
+export function invokeDetailedReportAPI(reportAPI, expectedOutputFile){
+    invokeReportAPI(reportAPI, expectedOutputFile, false);
+}
+
+export function invokeReportAPI(reportAPI, expectedOutputFile, isUsageReport){
+    cy.request(reportAPI).then((reportAPIResponse) => {
+        invokeLambdaFunction(reportAPIResponse.body, expectedOutputFile, isUsageReport);
+    });
+}
+
+function invokeLambdaFunction(inputData, expectedOutputFile, isUsageReport){
+
+    const USAGE_REPORT_LAMBDA = 'https://bn84q7k6u0.execute-api.us-east-1.amazonaws.com/Prod/create-query/?reportServiceSource=authoring.staging.concord.org&tokenServiceEnv=staging&usageReport=true';
+    const DETAILED_REPORT_LAMBDA = 'https://bn84q7k6u0.execute-api.us-east-1.amazonaws.com/Prod/create-query/?reportServiceSource=authoring.staging.concord.org&tokenServiceEnv=staging';
```

Comment:
Rather than hard coding these URLs it would be better if you could extract them from the page. 
One way of doing it could be:
https://github.com/abhinaba-ghosh/cypress-react-selector
Each button is a react component. 

This is a screenshot of what the the React chrome developer tools plugin shows for the "New Details Report" button.
![image](https://user-images.githubusercontent.com/86016/127868915-d5c1b11d-5df7-4dd1-9620-bee4ae2ac4f5.png)

I'd expect the cypress-react-selector would let the cypress tests see the same set of properties.

Without looking at the React virtual DOM getting this URL would be harder. The URLs are setup initially on the page here: https://github.com/concord-consortium/rigse/blob/9607c76abdc92c139221061517fbf70d403fb7b3/rails/app/views/report/learner/index.html.haml#L16

If you inspect the page you can see this script tag with the external reports being passed to the `PortalComponents.renderLearnerReportForm` function. I'm not sure the best way to get the URLs out of that.

Reply:
Will take a look at this plugin, Scott.

Reply:
@scytacki I will try to implement this plugin after this pull request is merged.
But I made the code easy to use the base url from button and then add filters from id. (currently hardcoded in test file)

Reply:
If we are happy with this framework i want to add more tests and merge this branch.